gateway/gateway.go

```diff
@@ -280,16 +280,14 @@ func (gw *Gateway) handleMessage(msg config.Message, dest *bridge.Bridge) []*BrM
 		return brMsgIDs
 	}
 
-	// Get the ID of the parent message in thread
-	var canonicalParentMsgID string
-	if msg.ParentID != "" && (gw.BridgeValues().General.PreserveThreading || dest.GetBool("PreserveThreading")) {
-		canonicalParentMsgID = gw.FindCanonicalMsgID(msg.Protocol, msg.ParentID)
-	}
-
 	originchannel := msg.Channel
 	origmsg := msg
 	channels := gw.getDestChannel(&msg, *dest)
 	for _, channel := range channels {
+		if parentID, isThreaded := gw.handleThreading(&msg, dest, channel); isThreaded {
```

Comment:
Why do we need a `isThreaded` bool here and in `handleThreading` ?
Is there a reason why we can't always give `msg.ParentID` a value?

Reply:
I think I was just copying the `handled bool` style that @Helcaraxon used, which I found very legible compared to setting based on opaque internals of function.

But if on consideration that's not helpful, I think it should be possible to set directly. Let me double-check on a complex gateway though :)